.github/workflows/ci.yml

```diff
@@ -7,6 +7,10 @@ on:
        - 'release/*'
    pull_request:
 
+concurrency:
+  group: ci-${{ github.head_ref }}
+  cancel-in-progress: true
+
```

Comment:
Neat.

Reply:
It seems to work and github.head_ref translates to the branch of the PR but what I don't know is how this will work on master, the doc at https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions says:

The head_ref or source branch of the pull request in a workflow run. This property is only available when the event that triggers a workflow run is either pull_request or pull_request_target.

Reply:
So the question becomes, what happens when that property is not available?

Reply:
Exactly.

Although the documentation for concurrency says to do exactly what I've done, but there are ways to configure a workflow to only run on PRs and not on master.

I can't see a way of making concurrency conditional either.

https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#concurrency